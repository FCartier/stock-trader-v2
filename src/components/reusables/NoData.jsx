import React from "react"

const NoData =({componentName}) => {
    return (
        <h5>No {componentName} currently available for this company.</h5>
    )
}

export default NoData;