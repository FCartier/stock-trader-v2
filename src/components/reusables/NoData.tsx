import * as React from "react";

interface NoDataProps {
    componentName: string,
}

const NoData: React.SFC<NoDataProps> = (props) => {
    return <p>No {props.componentName} currently available for this company.</p>
}

export function renderCondition(length: number) {
    if (length === 0) {
      return false;
    } else {
      return true;
    }
}
  
export default NoData;