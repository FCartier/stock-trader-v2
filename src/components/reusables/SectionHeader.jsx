import React from 'react'

export const SectionHeader = props => (
    <div className="category-header">
            <h3>{props.title}</h3>
    </div>
)