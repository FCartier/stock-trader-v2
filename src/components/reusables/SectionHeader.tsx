import * as React from "react";

import { CategoryHeader } from "./styles"
import { Link } from "react-router-dom"
import { connect } from "react-redux"

interface SectionHeaderProps {
    title: string
}

export const SectionHeader: React.SFC<SectionHeaderProps> = (props) => {
    return (
        <CategoryHeader>
           <h3>{props.title}</h3>
        </CategoryHeader>
    )
}

