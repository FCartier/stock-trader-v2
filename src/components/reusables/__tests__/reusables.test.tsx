import * as React from "react";
import { configure, shallow } from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import NoData from "../NoData";
import { SectionHeader } from "../SectionHeader";

configure({ adapter: new Adapter() });

describe("reusables", () => {
  test("NoData should render correctly", () => {
    const component = shallow(<NoData componentName="test" />)
    expect(component).toMatchSnapshot()
  })

  test("SectionHeader should render correctly", () => {
    const component = shallow(<SectionHeader title="Test" />)
    expect(component).toMatchSnapshot()
  })
})