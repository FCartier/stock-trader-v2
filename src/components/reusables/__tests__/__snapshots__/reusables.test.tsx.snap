// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reusables NoData should render correctly 1`] = `
<p>
  No 
  test
   currently available for this company.
</p>
`;

exports[`reusables SectionHeader should render correctly 1`] = `
<Styled(div)>
  <h3>
    Test
  </h3>
</Styled(div)>
`;
